src/Microsoft.AspNet.Mvc.Core/DefaultControllerTypeProvider.cs

```diff
@@ -76,17 +77,45 @@ protected internal virtual bool IsController([NotNull] TypeInfo typeInfo)
             {
                 return false;
             }
-            if (typeInfo.Name.Equals("Controller", StringComparison.OrdinalIgnoreCase))
+            if (typeInfo.Name.Equals(ControllerTypeName, StringComparison.OrdinalIgnoreCase))
             {
                 return false;
             }
-            if (!typeInfo.Name.EndsWith("Controller", StringComparison.OrdinalIgnoreCase) &&
-                !typeof(Controller).GetTypeInfo().IsAssignableFrom(typeInfo))
+            if (!typeInfo.Name.EndsWith(ControllerTypeName, StringComparison.OrdinalIgnoreCase) &&
```

Comment:
Fair, I just feel like its wasteful to do the check twice. Something that's not about derivation makes sense as a name then.